test/e2e/specs/editor/blocks/links.spec.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Links', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( `can be created by selecting text and using keyboard shortcuts`, async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		// Create a block with some text.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+		await page.keyboard.type( 'This is Gutenberg' );
+
+		// Select some text.
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft' );
+
+		// Press Cmd+K to insert a link.
+		await pageUtils.pressKeys( 'primary+K' );
+
+		// Type a URL.
+		await page.keyboard.type( 'https://wordpress.org/gutenberg' );
+
+		// Open settings.
+		await page.getByRole( 'button', { name: 'Link Settings' } ).click();
+
+		// Navigate to and toggle the "Open in new tab" checkbox.
+		const checkbox = page.getByLabel( 'Open in new tab' );
+		await checkbox.click();
+
+		// Toggle should still have focus and be checked.
+		await expect( checkbox ).toBeChecked();
+		await expect( checkbox ).toBeFocused();
+
+		// Ensure that the contents of the post have not been changed, since at
+		// this point the link is still not inserted.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+
+		// Tab back to the Submit and apply the link.
+		await page.getByRole( 'button', { name: 'Save' } ).click();
```

Comment:
This is failing. We should probably limit the scope under some parent locator. I found out that the link setting popover doesn't have an accessible name. When I'm testing it, I also noticed that the focus is trapped inside the popover, so maybe what it should really be is a modal/dialog instead? This can all be a follow-up though 😅.

Reply:
Good point, I could only target the parent using a class. Let's follow up by improving the popover and updating the test as necessary